Mathlib/Algebra/Module/Basic.lean

```diff
@@ -300,6 +301,12 @@ theorem sub_smul (r s : R) (y : M) : (r - s) • y = r • y - s • y := by
   simp [add_smul, sub_eq_add_neg]
 #align sub_smul sub_smul
 
+lemma Int.units_smul_smul (r : ℤˣ) (s : R) (m : M) : r • (s • m) = (r • s) • m := by
+  obtain rfl | rfl := Int.units_eq_one_or r <;> simp
+
+lemma Int.smul_units_smul (r : R) (s : ℤˣ) (m : M) : r • (s • m) = (s • r) • m := by
+  obtain rfl | rfl := Int.units_eq_one_or s <;> simp
```

Comment:
These should both just be applications of `smul_comm`; maybe an instance is missing (or in another file)?

Reply:
I am not sure it is that much related to `SMulCommClass`. (I think `smul_comm` would only be able to reduce the second lemma to the first, and vice versa, but each individual lemma is independent from it.)

Reply:
It might be related to `IsScalarTower` though...

Reply:
Ah, indeed; I think this statement is simply the combination of `smul_comm` and `smul_assoc`, unless I'm missing something.